src/dashboard/src/main/models.py

```diff
@@ -551,12 +550,28 @@ class Job(models.Model):
     currentstep = models.IntegerField(db_column='currentStep', choices=STATUS, default=0, blank=False)
     microservicegroup = models.CharField(max_length=50, db_column='microserviceGroup', blank=True)
     hidden = models.BooleanField(default=False)
-    microservicechainlink = models.ForeignKey('MicroServiceChainLink', db_column='MicroServiceChainLinksPK', null=True, blank=True)
+    microservicechainlink = UUIDField(auto=False, null=True, blank=True, db_column='MicroServiceChainLinksPK')
     subjobof = models.CharField(max_length=36, db_column='subJobOf', blank=True)
 
     class Meta:
         db_table = u'Jobs'
 
+    def get_directory_name(self, default=None):
+        if not self.directory:
+            return self.sipuuid
+        try:
+            return re.search(
+                r'^.*/(?P<directory>.*)-'
+                r'[\w]{8}(-[\w]{4})'
+                r'{3}-[\w]{12}[/]{0,1}$', self.directory).group('directory')
+        except Exception:
```

Comment:
I've added a docstring trying to explain why we were doing this. Maybe there's more to it than I'd been able to capture. I think that it was used generally because `Job` is a single entity common to different unit types. A lot of views use it but then I think it could also be done differently. It feels like `Job` shouldn't be responsible for that.